### PR TITLE
Fix controller instantiation in UXD itxs

### PR DESCRIPTION
**Pull request by sjillen**
pass an actual instance of the controller instead of just the pda

### CODE REVIEW

File: tools/sdk/uxdProtocol/createWithdrawInsuranceFromMangoDepositoryInstruction.ts [hunk 1]
```diff
@@ -31,7 +30,7 @@ const createWithdrawInsuranceFromMangoDepositoryInstruction = async (
 
   return client.createWithdrawInsuranceFromMangoDepositoryInstruction(
     insuranceWithdrawnAmount,
-    { pda: getControllerPda(uxdProgramId) } as Controller,
+    instantiateController(uxdProgramId),
```
Comment (acamill) [inline tools/sdk/uxdProtocol/createWithdrawInsuranceFromMangoDepositoryInstruction.ts:33]: Any other place in the code using a mocked Controller or Depository?

**Review by acamill — APPROVED** [review]

**Comment by acamill** [comment]
When you merge this one, let's prep the daily rais cap of 250k
Screenshot 2022-01-23 at 23 32 20

File: tools/sdk/uxdProtocol/createWithdrawInsuranceFromMangoDepositoryInstruction.ts (hunk 1 above)
Comment (sjillen) [reply tools/sdk/uxdProtocol/createWithdrawInsuranceFromMangoDepositoryInstruction.ts:33]: I only used it in a few `create*Instruction` files of ours, it's all been replaced be `instantiateController`.

File: tools/sdk/uxdProtocol/uxdClient.ts [hunk 2]
```diff
@@ -139,10 +139,9 @@ export const getControllerPda = (uxdProgramId: PublicKey): PublicKey => {
   return findAddrSync([Buffer.from('CONTROLLER')], uxdProgramId)[0]
 }
 
-// We do not need the mint symbol so it is just set with a placeholder value
 export const instantiateController = (
   uxdProgramId: PublicKey,
-  mintDecimals: number,
+  mintDecimals = 6,
```
Comment (0xCryptoSheik) [inline tools/sdk/uxdProtocol/uxdClient.ts:144]: Ok as long as we're very careful with these defaults

Comment (acamill) [reply tools/sdk/uxdProtocol/uxdClient.ts:144]: Missed this one. It's in UXDClient as UXD_DECIMALS.

https://github.com/UXDProtocol/uxd-client/blob/9e521a7da5a80c9f445c3ea0d9b5c81b927ccfce/src/utils.ts#L19

There should be no hardcoded value. If it's not in the UXD_CLIENT then it should be added.

Comment (acamill) [reply tools/sdk/uxdProtocol/uxdClient.ts:144]: Also bellow I see depository decimal to 6? Why. Why put default values that make no sense, it's easy to overlook later.

Rather have it onsite where it's use with either the proper value or "value not used, passing DEFAULT_DECIMALS or something. Tiny things but put all together it become quite hard to reason about and overlook issues.

Screenshot 2022-01-24 at 10 39 12

Comment (acamill) [reply tools/sdk/uxdProtocol/uxdClient.ts:144]: Would rather get rid of the `export const instantiateMangoDepository = (` and use the original one. Gain is so small, and possible obfuscation/confusion later on makes it not worth imo

Comment (acamill) [reply tools/sdk/uxdProtocol/uxdClient.ts:144]: Same for `export const instantiateController = (`

**Pull request merged**